src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -959,14 +904,35 @@ void Graph::PullOutputData(BlobMap &out) {
 
         auto srcPrec = actualDesc.getPrecision();
         auto dstPrec = expectedDesc.getPrecision();
-
-        if (!getConfig().isLegacyApi && srcPrec == dstPrec && ext_blob->byteSize() != intr_blob.GetSize())
-            IE_THROW() << "Output blob byte size is not equal network output byte size (" << ext_blob->byteSize()
+        if (!getConfig().isLegacyApi && srcPrec == dstPrec && ext_blob.get_byte_size() != intr_blob.GetSize())
+            IE_THROW() << "Output blob byte size is not equal network output byte size (" << ext_blob.get_byte_size()
                        << "!=" << intr_blob.GetSize() << ").";
 
-        void *ext_blob_ptr = ext_blob->buffer();
+        void *ext_blob_ptr = ext_blob.data();
         void *intr_blob_ptr = intr_blob.GetData();
 
+        // If output precision has been changed comparing to original model's output, it must be copied to aux tensor
+        if (out_precision_changed[name]) {
+            auto it = aux_tensors.find(name);
+            if (it == aux_tensors.end()) {
+                OPENVINO_THROW("Output precision has been changed, but cannot find its aux tensor.");
+            }
+
+            auto& aux_tensor = it->second;
+            aux_tensor.set_shape(out[name].get_shape());
+            ext_blob_ptr = aux_tensor.data();
+            if ((intr_blob_ptr == nullptr) || (ext_blob_ptr == nullptr)) {
+                OPENVINO_THROW("Get tensor has no allocated memory");
+            }
+
+            cpu_convert(intr_blob_ptr,
+                        ext_blob_ptr,
+                        srcPrec,
+                        InferenceEngine::details::convertPrecision(aux_tensor.get_element_type()),
+                        intr_blob.GetDescWithType<BlockedMemoryDesc>()->getPaddedElementsCount());
+            continue;
+        }
+
```

Comment:
Above code(L915-L935) is used to handle the precision change between original model and compiled model, if the precision has changed, it will do precision conversion. While L939-L948 will check BlockingDesc difference, which contains precision/layout/stride/shape information. We need notice that ov::Tensor has no layout info, only contains precision/shape/stride info, so we just can check stride/shape info's difference in L939.
So we check:  if only precision change we go to code(L915-L935), else shape/stride changes then go to L939.

BTW, If user set a tensor which is incompatible with its port, it will report error before inference.

Reply:
@ceciliapeng2011 Does it always assume that precision not change when reorder (L939-L948) ? Because it seems that it doesn't resize the buffer size, while precision change will lead to data byte size be changed, right?
We also can see that `actualDesc.getBlockingDesc() != expectedDesc.getBlockingDesc()` will only check shape/layout/stride:
https://github.com/openvinotoolkit/openvino/blob/8f529df763f6bba38af85748d8d24f495bb673f1/src/inference/src/ie_layouts.cpp#L417-L420

Anyway, in plugin API 2.0 there is no layout info in ov::Tensor, so seems that we cannot compare the layout any more. Maybe shape can be used to replace layout in reorder operation. So in the case of precision change + reorder, we can first do precision conversion and then do reoder.